### DISCUSSION TITLE
Add listener to GlideImage

### DISCUSSION THREAD

**Discussion by kaychyka17** [skydoves/landscapist#69] [Q&A] [answered]
If I noticed correctly there is no possibility to add listener to GlideImage. I'm loading .svg from url and for that SVG listener is needed. Is there any chance this will be added soon?

**Comment by skydoves** [comment, marked as answer]
Hello, you can set the [RequestBuilder](https://github.com/skydoves/landscapist#requestbuilder) as following the documentation.

```kotlin
  GlideImage(
    imageModel = poster,
    requestBuilder = Glide.with(LocalContext.current.applicationContext).asDrawable()
      .listener(object : RequestListener<Drawable> {
        override fun onLoadFailed(
          e: GlideException?,
          model: Any?,
          target: Target<Drawable>?,
          isFirstResource: Boolean
        ): Boolean {
          TODO("Not yet implemented")
        }

        override fun onResourceReady(
          resource: Drawable?,
          model: Any?,
          target: Target<Drawable>?,
          dataSource: DataSource?,
          isFirstResource: Boolean
        ): Boolean {
          TODO("Not yet implemented")
        }
      }),
```

**Reply by jennatauro** [reply]
@skydoves Hi, am I missing something?

I am using the following code and the `onResourceReady` callback is never called.

```
GlideImage(
        imageModel = imageModel.imageUrl,
        requestBuilder = { Glide.with(LocalContext.current.applicationContext).asDrawable()
            .listener(object : RequestListener<Drawable> {
                override fun onLoadFailed(
                    e: GlideException?,
                    model: Any?,
                    target: Target<Drawable>?,
                    isFirstResource: Boolean
                ): Boolean {
                    return false
                }

                override fun onResourceReady(
                    resource: Drawable?,
                    model: Any?,
                    target: Target<Drawable>?,
                    dataSource: DataSource?,
                    isFirstResource: Boolean
                ): Boolean {
                    return false
                }
            }) },
        contentDescription = imageModel.contentDescription,
        contentScale = ContentScale.Fit,
        modifier = Modifier
            .clip(CircleShape)
            .size(dimensionResource(id = avatarSize.diameter))
            .background(colorResource(id = R.color.hover_blue))
    )
```

The images are loaded and everything, it's just the callbacks are not called. I don't see any errors either.

**Reply by skydoves** [reply]
Hi @jennatauro, I'm generally recommend using [custom Composables](https://github.com/skydoves/landscapist#custom-composables), but I'll take a look at the custom `requestBuilder`.

**Reply by jennatauro** [reply]
@skydoves do you mean to use the `success` parameter? I don't want to show something custom on success though, I want to let the glide image be displayed as it currently is but I want to log something on successful image load.

How can I accomplish this using custom Composables?

**Reply by skydoves** [reply]
@jennatauro Thank you for the details! Just released `1.4.9-SNAPSHOT` on the maven central repository.
Please check out the codes below work well:

```kotlin
  GlideImage(
    imageModel = poster.poster,
    modifier = Modifier
      .aspectRatio(0.8f),
    requestListener = object : RequestListener<Drawable> {
      override fun onLoadFailed(
        e: GlideException?,
        model: Any?,
        target: com.bumptech.glide.request.target.Target<Drawable>?,
        isFirstResource: Boolean
      ): Boolean {
        return false
      }

      override fun onResourceReady(
        resource: Drawable?,
        model: Any?,
        target: Target<Drawable>?,
        dataSource: DataSource?,
        isFirstResource: Boolean
      ): Boolean {
        return false
      }
    },
..
)
```

**Reply by jennatauro** [reply]
@skydoves thank you for this change! Do you plan to release this change officially anytime soon? Our application is user-facing and we don't use snapshot versions of libraries.

**Reply by jennatauro** [reply]
Also, I am not able to find `1.4.9-SNAPSHOT` in the maven central repository

**Reply by skydoves** [reply]
@jennatauro You can import the snapshot following [this guide](https://github.com/skydoves/landscapist#snapshot).
If you find it works, the next stable will be released very soon (probably today). Thanks!

**Reply by jennatauro** [reply]
@skydoves oh it works, thank you!!

**Reply by jennatauro** [reply]
Hi @skydoves would it actually be possible for you to add this functionality to `CoilImage` as well? I've just found out my team wants to move over to using Coil instead of Glide but this functionality solved a big testing problem for us so I would love to request that and a release with that change. Thank you!

**Reply by skydoves** [reply]
Hi @jennatauro, the new version `1.5.0-SNAPSHOT` has been released. Please check out the version work well for you.
You can use it as seen in the below:

```kotlin
    CoilImage(
        requestListener = object : ImageRequest.Listener {
            override fun onStart(request: ImageRequest) {
                // ..
            }

            override fun onSuccess(request: ImageRequest, metadata: ImageResult.Metadata) {
                // ..
            }
        },
```

**Reply by jennatauro** [reply]
@skydoves yes this works. Thank you for the fast reply! Would love to use this as a new released version :)

**Reply by jennatauro** [reply]
@skydoves could you add an official release for this change when you get the chance? Thank you!

**Reply by skydoves** [reply]
Hi, @jennatauro a new stable `1.5.0` has been released. And I was wondering if you don't mind, could you share the name of your team's service? I would love to know and share it with communities how this library is being used in real-world projects. Thank you!

**Reply by jennatauro** [reply]
@skydoves thank you! Yes, I was actually hoping to add us to your usecases doc here: https://github.com/skydoves/landscapist/blob/main/usecases.md
Our team is [Faire Wholesale](https://www.faire.com/?utm_source=google&utm_medium=cpc&utm_campaign=SEM_AW_AMER_CA___RTL_BR___EXT&keyword=faire%20wholesale&matchtype=e&device=c&adgroupid=74190523370&gclid=Cj0KCQiAybaRBhDtARIsAIEG3klMN0fdYUEp5eD0hfJZEFlTmbhbkKGz3Gv1mhy2l1XqAXqDO3rxR3AaAjcrEALw_wcB), this is our Android app https://play.google.com/store/apps/details?id=com.faire.retailer&hl=en_CA&gl=US

Let me know if there's anything you need from me to include our app as one of your usecases! Thank you

**Reply by skydoves** [reply]
Thank you for letting me know, @jennatauro! I love your team's app! I'll definitely add the app to the use case doc. Thank you! :)
